account_reconcile_model_multicompany_propagate/models/account_reconcile_model.py

```diff
@@ -0,0 +1,183 @@
+                    try:
+                        target_account = accounts_map[company.id][line.account_id.code]
+                    except KeyError:
+                        _logger.warning(
+                            "Not propagating account to company because it does "
+                            "not exist there: company=%s, account=%s",
+                            company,
+                            line.account_id.code,
+                        )
+                        continue
+                    target_taxes_ids = []
+                    for tax in line.tax_ids:
+                        try:
+                            target_taxes_ids.append(taxes_map[company.id][tax.name])
+                        except KeyError:
+                            _logger.warning(
+                                "Not propagating tax to company because it does "
+                                "not exist there: company=%s, tax=%s",
+                                company,
+                                tax.name,
+                            )
+                            continue
+                    rec_model_lines_ids.append(
+                        self.env["account.reconcile.model.line"]
+                        .create(
+                            {
+                                "model_id": rec_model.id,
+                                "sequence": line.sequence,
+                                "account_id": target_account,
+                                "label": line.label,
+                                "amount_type": line.amount_type,
+                                "force_tax_included": line.force_tax_included,
+                                "amount_string": line.amount_string,
+                                "tax_ids": [(6, 0, target_taxes_ids)],
+                            }
+                        )
+                        .id
+                    )
+                rec_model.update(
+                    {
+                        "line_ids": [(6, 0, rec_model_lines_ids)],
+                    }
+                )
+
+            # Update partner mapping lines
+            if self.partner_mapping_line_ids:
+                rec_model_partners = self.env[
+                    "account.reconcile.model.partner.mapping"
+                ].create(
+                    [
+                        {
+                            "model_id": rec_model.id,
+                            "partner_id": line.partner_id.id,
+                            "payment_ref_regex": line.payment_ref_regex,
+                            "narration_regex": line.narration_regex,
+                        }
+                        for line in self.partner_mapping_line_ids
+                    ]
+                )
+                rec_model.update(
+                    {
+                        "partner_mapping_line_ids": [(6, 0, rec_model_partners.ids)],
+                    }
+                )
+
+    def _get_vals_create(self, company):
```

Comment:
Also a small rename to make the method's purpose more obvious:

```suggestion
    def _get_vals_create_propagated(self, company):
```

Reply:
Method renamed